DATA RACE in CI

Here is the detail:

<details>

```
[2020-03-11T13:01:50.713Z] ==================
[2020-03-11T13:01:50.713Z] WARNING: DATA RACE
[2020-03-11T13:01:50.713Z] Write at 0x00c00084cf30 by goroutine 1394:
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server/region_syncer.(*RegionSyncer).establish()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/region_syncer/client.go:94 +0x55d
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server/region_syncer.(*RegionSyncer).StartSyncWithLeader.func1()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/region_syncer/client.go:137 +0x393
[2020-03-11T13:01:50.713Z] 
[2020-03-11T13:01:50.713Z] Previous read at 0x00c00084cf30 by goroutine 1167:
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server/region_syncer.(*RegionSyncer).reset()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/region_syncer/client.go:52 +0x89
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server/region_syncer.(*RegionSyncer).StopSyncWithLeader()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/region_syncer/client.go:40 +0x3c
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server.(*Server).leaderLoop()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:1050 +0xa5e
[2020-03-11T13:01:50.713Z] 
[2020-03-11T13:01:50.713Z] Goroutine 1394 (running) created at:
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server/region_syncer.(*RegionSyncer).StartSyncWithLeader()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/region_syncer/client.go:122 +0xc2
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server.(*Server).leaderLoop()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:1046 +0xb2d
[2020-03-11T13:01:50.713Z] 
[2020-03-11T13:01:50.713Z] Goroutine 1167 (running) created at:
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server.(*Server).startServerLoop()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:472 +0x106
[2020-03-11T13:01:50.713Z]   github.com/pingcap/pd/v4/server.(*Server).Run()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:454 +0x167
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x284
[2020-03-11T13:01:50.713Z]   fmt.(*ss).doScanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:1217 +0x382
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x192
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb.(*DB).checkAndCleanFiles()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db_util.go:52 +0x2f8
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb.openDB()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:130 +0x9f0
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x192
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x192
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x284
[2020-03-11T13:01:50.713Z]   fmt.(*ss).doScanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:1217 +0x382
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:114 +0x192
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb.(*DB).recoverJournal()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:476 +0xc5
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb.openDB()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:125 +0x9cc
[2020-03-11T13:01:50.713Z]   github.com/syndtr/goleveldb/leveldb.Open()
[2020-03-11T13:01:50.713Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:197 +0x1c6
[2020-03-11T13:01:50.713Z]   fmt.Fscanf()
[2020-03-11T13:01:50.713Z]       /usr/local/go/src/fmt/scan.go:143 +0xcf
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
[2020-03-11T13:01:50.714Z]       /usr/local/go/src/fmt/scan.go:114 +0x192
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb.(*session).recover.func1()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/session.go:113 +0xf4
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb.(*session).recover()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/session.go:121 +0x1929
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb.Open()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:183 +0xf8
[2020-03-11T13:01:50.714Z]   github.com/syndtr/goleveldb/leveldb.OpenFile()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:219 +0xa5
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/server/kv.NewLeveldbKV()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/kv/levedb_kv.go:31 +0x52
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/server/core.NewRegionStorage()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/core/region_storage.go:53 +0x5d
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/server.(*Server).startServer()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:361 +0xd00
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/server.(*Server).Run()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/server.go:450 +0xfd
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/tests.(*TestServer).Run()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/tests/cluster.go:100 +0xec
[2020-03-11T13:01:50.714Z]   github.com/pingcap/pd/v4/tests.(*TestCluster).RunServer.func1()
[2020-03-11T13:01:50.714Z]       /home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/tests/cluster.go:353 +0x38
[2020-03-11T13:01:50.714Z] ==================

...

[2020-03-11T13:01:50.729Z] --- FAIL: Test (126.39s)
[2020-03-11T13:01:50.729Z]     testing.go:853: race detected during execution of test
[2020-03-11T13:01:50.729Z] FAIL
[2020-03-11T13:01:50.729Z] coverage: [no statements]
[2020-03-11T13:01:50.729Z] FAIL	github.com/pingcap/pd/v4/tests/server/member	126.645s
```

</details>

--- a/server/region_syncer/client.go
+++ b/server/region_syncer/client.go
@@ -91,7 +91,9 @@ func (s *RegionSyncer) establish(addr string) (*grpc.ClientConn, error) {
 		return nil, errors.WithStack(err)
 	}
 
+	s.Lock()
 	s.regionSyncerCtx, s.regionSyncerCancel = ctx, cancel
+	s.Unlock()
 	return cc, nil
 }
 

